samza-core/src/main/java/org/apache/samza/operators/MessageStreamImpl.java

```diff
@@ -72,54 +74,62 @@ public MessageStreamImpl(StreamGraphImpl graph, OperatorSpec<?, M> operatorSpec)
 
   @Override
   public <TM> MessageStream<TM> map(MapFunction<? super M, ? extends TM> mapFn) {
-    OperatorSpec<M, TM> op = OperatorSpecs.createMapOperatorSpec(mapFn, this.graph.getNextOpId());
+    String opId = this.graph.getNextOpId(OpCode.MAP, null);
```

Comment:
minor: it seems that getNextOpId(code, null) was repeated many times. Might be better to add another getNextOpId(code) to avoid typing null repeatedly.  But, you call.